### PR TITLE
add MetainfoRelatedValueFilter (EDM-304, EDM-204)

### PR DESCRIPTION
**Pull request by dmitry-at-genestack**
Added support for search by related 
See [EDM-304](https://genestack.atlassian.net/browse/EDM-304).

Merge after https://github.com/genestack/genestack-applications/pull/443 is merged.

### CODE REVIEW

File: genestack_client/file_filters.py [hunk 1]
```diff
@@ -185,3 +185,21 @@ class OrFileFilter(FileFilter):
     def __init__(self, first, second):
         super(OrFileFilter, self).__init__()
         self._dict.update({'or': [first.get_dict(), second.get_dict()]})
+
+
+class MetainfoRelatedValueFilter(FileFilter):
+    """
+    Filter to select files with metainfo values related to given terms by a dictionary relationship.
+    """
+    def __init__(self, key, term_labels, dictionary_accession, relationship_label, transitive):
```
Comment (genestack-solomatin) [inline genestack_client/file_filters.py]: Can you add docstrings about arguments. 

Some examples of the `relationship_label` and description of what transitive do.

File: genestack_client/file_filters.py [hunk 2]
```diff
@@ -185,3 +185,21 @@ class OrFileFilter(FileFilter):
     def __init__(self, first, second):
         super(OrFileFilter, self).__init__()
         self._dict.update({'or': [first.get_dict(), second.get_dict()]})
+
+
+class MetainfoRelatedValueFilter(FileFilter):
+    """
+    Filter to select files with metainfo values related to given terms by a dictionary relationship.
```
Comment (alexander-polovtcev) [inline genestack_client/file_filters.py]: This API is actually 'more' public than the Java API, so it its docsting must be at least as big as the javadoc. I'd also add some usage examples.

File: genestack_client/file_filters.py (hunk 1 above)
Comment (dmitry-at-genestack) [reply genestack_client/file_filters.py]: Done

File: genestack_client/file_filters.py [hunk 3]
```diff
@@ -185,3 +185,41 @@ class OrFileFilter(FileFilter):
...
+
+    For example, if dictionary contains relationship "broader" and terms are related as:
+        - "Nervous system" is broader than "Brain"
+        - "Brain" is broader than "Midbrain"
+    Only files with "Midbrain" will be found by "Brain" query with `transitive==false`.
+    With `transitive==true` both "Brain" and "Nervous system" queries will find these files.
+    """
+    def __init__(self, key, term_labels, dictionary_accession, relationship_label, transitive):
+        """
+        :param key: metainfo key
```
Comment (alexander-polovtcev) [inline genestack_client/file_filters.py]: Add type specification for parameters

Comment (dmitry-at-genestack) [reply genestack_client/file_filters.py]: Done

**Pull request merged**